### PR TITLE
Add debugging tools tutorial

### PR DESCRIPTION
**Pull request by gabrielfruet**
## What has changed and why?

Added a debugging tools tutorial that walks through detecting and
diagnosing gradient overflow with `underflow_overflow` and gradient
norm logging. Uses the fine-tuning API (pretraining doesn't accept
`debug_args`) with real captured outputs from a deterministic
synthetic dataset — no placeholder values.

## How has it been tested?

Smoke-tested the full path in a fresh shell: `setup_data.py`, the
three training scripts, and the gradient-norm visualization. All
complete and match the outputs shown in the tutorial.

## Did you update CHANGELOG.md?

- [x] Yes

## Did you update the documentation?

- [x] Yes

### CODE REVIEW

File: docs/source/tutorials/debugging_tools/index.md [hunk 1]
```diff
@@ -0,0 +1,357 @@
...
+```{note}
+The debugging tools (`debug_args`) are only available in the fine-tuning
+commands (`train_image_classification`, `train_object_detection`, …), not in
+self-supervised `pretrain`. This tutorial uses `train_image_classification`.
+```
+
+## Install Dependencies
+
+```bash
+pip install lightly-train torch torchvision matplotlib
```
Comment (liopeer) [inline docs/source/tutorials/debugging_tools/index.md]:
```suggestion
pip install lightly-train
```
All of these are core dependencies anyway.

File: docs/source/tutorials/debugging_tools/index.md [hunk 2]
```diff
@@ -0,0 +1,357 @@
...
+CLASS_NAMES = [
+    "airplane", "automobile", "bird", "cat", "deer",
+    "dog", "frog", "horse", "ship", "truck",
+]
+DATA_DIR = Path("datasets/debugging_tutorial")
+NUM_TRAIN_PER_CLASS = 32
+NUM_VAL_PER_CLASS = 8
+IMAGE_SIZE = (96, 96)
+
+for split, num_per_class in (("train", NUM_TRAIN_PER_CLASS), ("val", NUM_VAL_PER_CLASS)):
```
[the remark this answers is not in the document]
Comment (liopeer) [reply docs/source/tutorials/debugging_tools/index.md]: This does indeed not sound optimal. Would just redefine the class names in the other script.

File: docs/source/tutorials/debugging_tools/index.md [hunk 3]
```diff
@@ -0,0 +1,357 @@
...
+        out_fp16 = torch.where(
+            x_fp16 > 0,
+            torch.exp(x_fp16 * self.scale),
+            torch.zeros_like(x_fp16),
+        )
+        return out_fp16.to(dtype=x.dtype)
+
+
+@contextmanager
+def patched_resnet18(scale: float = 1.0):
```
Comment (liopeer) [inline docs/source/tutorials/debugging_tools/index.md:95]: Damn, I thought it was possible to just pass your own torch models into the `train_image_classification` (as long as they implement the required methods from the wrapper), which would have made this a lot simpler.

**Review by liopeer — APPROVED** [review]
LGTM! We can maybe make it a bit easier in the future if we ever allow passing custom models (which we should and I thought we actually do).
